types/yuka/tslint.json

```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-unnecessary-class": false
+    }
```

Comment:
Can this be removed?

Reply:
We have a bunch of all static classes in the original code base.

Namely these are:
[types/yuka/src/core/Logger.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52724/files#diff-ab34d9bd6f2e8c9c1e81aef48bc38fe0284f77afc75a3882498d525dd22e0734) see [Original here](https://github.com/Mugen87/yuka/blob/bd9430fe2131a875e6c9250d0ded7a04eaab52f7/src/core/Logger.js)
[types/yuka/src/graph/extra/HeuristicPolicy.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52724/files#diff-ba08db1a3d9a87e3ea22c8145d92462b1c805ed66bc486a1ccb83106a2761c9d) see [Original here](https://github.com/Mugen87/yuka/blob/bd9430fe2131a875e6c9250d0ded7a04eaab52f7/src/graph/extra/HeuristicPolicy.js)
[types/yuka/src/math/MathUtils.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52724/files#diff-6f395cb7d66f0e3b367ec34617e9f11e6c47ca570c300ad0f0c621062746b2d8) see [Original here](https://github.com/Mugen87/yuka/blob/bd9430fe2131a875e6c9250d0ded7a04eaab52f7/src/math/MathUtils.js)
[types/yuka/src/navigation/extra/GraphUtils.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52724/files#diff-79a03b14f94589a04fdb6e195ae07c0f61951e53907658ed5bf0b63f8092401c) see [Original here](https://github.com/Mugen87/yuka/blob/bd9430fe2131a875e6c9250d0ded7a04eaab52f7/src/graph/extra/HeuristicPolicy.js)

As they all are real ES6 classes in the original codebase, I decided to keep them classes.
If I am wrong on this, how should I rework them?
If you would guide me how to express these better, this can be removed.

Reply:
I think that [this](https://www.reddit.com/r/typescript/comments/8oafqb/class_with_all_static_methods_vs_namespace_vs/e0cytl3/) is a good advice; but you could also just type it as an object if it's not instantiated, no?

In any case, :shrug:

Reply:
> I think that [this](https://www.reddit.com/r/typescript/comments/8oafqb/class_with_all_static_methods_vs_namespace_vs/e0cytl3/) is a good advice; but you could also just type it as an object if it's not instantiated, no?
> 
> In any case,

When designing the system I totally agree with you, the js code should not define all static classes but rather a sub module or the like. Even a singleton would be better.
Yet, these are type definitions for said existing code, wouldn't it mask the original structure and therefore open up a can of worms for people when I change this to module/namespace in the types?
My feeling is, we should rather keep them as is.